pkg/entities/record.go

```diff
@@ -33,12 +35,13 @@ import (
 
 type Record interface {
 	PrepareRecord() (uint16, error)
-	AddInfoElement(element *InfoElement, val interface{}) (uint16, error)
+	AddInfoElement(element *InfoElement, val interface{}, isDecoding bool) (uint16, error)
 	// TODO: Functions for multiple elements as well.
 	GetBuffer() *bytes.Buffer
 	GetTemplateID() uint16
 	GetFieldCount() uint16
 	GetTemplateElements() []*InfoElement
+	GetDataElements() []*InfoElementValue
```

Comment:
If value is added to InfoElement, we can have one interface GetInfoElements that works for both data record and template record?

Reply:
Sounds good. Will do that. thanks!